contracts/SortedVessels.sol

```diff
@@ -394,23 +385,20 @@ contract SortedVessels is OwnableUpgradeable, ISortedVessels {
 		uint256 _NICR,
 		address _startId
 	) internal view returns (address, address) {
+		Data storage assetData = data[_asset];
+
 		// If `_startId` is the tail, check if the insert position is after the tail
-		if (
-			data[_asset].tail == _startId && _NICR <= _vesselManager.getNominalICR(_asset, _startId)
-		) {
+		if (assetData.tail == _startId && _NICR <= _vesselManager.getNominalICR(_asset, _startId)) {
 			return (_startId, address(0));
 		}
 
 		address nextId = _startId;
-		address prevId = data[_asset].nodes[nextId].prevId;
+		address prevId = assetData.nodes[nextId].prevId;
 
 		// Ascend the list until we reach the end or until we find a valid insertion point
-		while (
-			nextId != address(0) &&
-			!_validInsertPosition(_asset, _vesselManager, _NICR, prevId, nextId)
-		) {
-			nextId = data[_asset].nodes[nextId].prevId;
-			prevId = data[_asset].nodes[nextId].prevId;
+		while (nextId != address(0) && !_validInsertPosition(_asset, _vesselManager, _NICR, prevId, nextId)) {
+			nextId = assetData.nodes[nextId].prevId;
+			prevId = assetData.nodes[nextId].prevId;
```

Comment:
Same here.

Reply:
Nope, look closely and you'll notice that prevId changes between those two lines, so we cannot cache the lookup.